examples/cylinderflow.py

```diff
@@ -58,32 +58,33 @@ def main(nelems: int, degree: int, reynolds: float, rotation: float, timestep: f
     ns.define_for('x', gradient='∇', normal='n', jacobians=('dV', 'dS'))
     J = ns.x.grad(geom)
     detJ = function.determinant(J)
-    ns.ubasis = function.matmat(function.vectorize([
+    ubasis = function.matmat(function.vectorize([
         domain.basis('spline', degree=(degree, degree-1), removedofs=((0,), None)),
         domain.basis('spline', degree=(degree-1, degree))]), J.T) / detJ
-    ns.pbasis = domain.basis('spline', degree=degree-1) / detJ
-    ns.u = function.dotarg('u', ns.ubasis)
-    ns.p = function.dotarg('p', ns.pbasis)
+    pbasis = domain.basis('spline', degree=degree-1) / detJ
+    ns.u = function.dotarg('u', ubasis)
+    ns.v = function.dotarg('v', ubasis)
+    ns.p = function.dotarg('p', pbasis)
+    ns.q = function.dotarg('q', pbasis)
```

Comment:
I think we can use `ns.add_field` here as well.

Reply:
fixed